### PR TITLE
implement service entry SANs matching

### PR DESCRIPTION
**Pull request by GregHanson**

### CODE REVIEW

**Comment by GregHanson** [comment]
dependent on https://github.com/kdorosh/istio/pull/1 and also https://github.com/istio/istio/pull/45471

File: src/tls/boring.rs [hunk 1]
```diff
@@ -386,7 +383,7 @@ enum Verifier {
     None,
 
     // Allows exactly one identity, making sure at least one of the presented certs matches that identity
-    San(Identity),
+    San(Vec<Identity>),
```
Comment (hzxuzhonghu) [inline src/tls/boring.rs:386]: update comment as well

**Review by hzxuzhonghu — COMMENTED** [review]
lgtm

File: src/state.rs [hunk 2]
```diff
@@ -40,6 +40,7 @@ pub mod workload;
 pub struct Upstream {
     pub workload: Workload,
     pub port: u16,
+    pub sans: Vec<String>,
```
Comment (ymesika) [inline src/state.rs:43]: If this is always empty vector (didn't see any place where san is being added to it) why do you need it?

**Review by ymesika — COMMENTED** [review]
Left a comment but overall lgtm.

File: src/state.rs (hunk 2 above)
Comment (GregHanson) [reply src/state.rs:43]: thanks - this snuck in during that latest merge with latest

**Review by ymesika — APPROVED** [review]
lgtm

**Review by howardjohn — APPROVED** [review]

**Review by kdorosh — COMMENTED** [review]
how did we test that this works? i don't see any tests added covering the new behavior AFAICT

**Comment by GregHanson** [comment]
follow-up PR to add some tests here:
https://github.com/istio/ztunnel/pull/601

**Pull request merged**